src/components/subscribe/SubscriptionBox.vue

```diff
@@ -42,17 +49,30 @@ export default {
       if (this.isLoggedIn) {
         selectSubscriptionPlan(planType)
         // Redirect to subscription cc page:
-        this.$router.replace('/subscribe')
+        this.$router.push('/subscribe')
       } else {
         // Redirect to  sign up page, then redirect to Subscription
         // OR at least show a banner
         selectSubscriptionPlan(planType)
-        this.$router.replace('/register')
+        this.$router.push('/register')
       }
     }
   },
   computed: {
-    ...mapGetters(['isLoggedIn'])
+    ...mapGetters(['isLoggedIn']),
+    ...mapState({
+      alreadySubscribed(state) {
+        if (!this.isLoggedIn) return false;
+        if (state.me && state.me.subscription && state.me.subscription.active) {
+          return true;
+        } else {
+          return false;
+        }
+      },
```

Comment:
Is this mapsState required? Creating it inside computed directly will generate a state-like computed property. Don't?

Reply:
Good catch. I saw it used for similar logic elsewhere and now that I'm looking through it more closely, it can definitely be cleaned up.